pkg/controller/machineset/metrics.go

```diff
@@ -0,0 +1,69 @@
+package machineset
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/klog"
+)
+
+var (
+	// FailedMachineSetProvisionCount calculates the number of failures for a specific MachineSet
+	FailedMachineSetProvisionCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "failed_machine_sets_total",
+			Help: "Number of times machine set provisioning has failed.",
+		}, []string{"name", "namespace", "reason"},
+	)
+
+	// SucceededMachineSetProvisionCount calculates the number of success provisioning for a MachineSet
+	SucceededMachineSetProvisionCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "succeeded_machine_sets_total",
+			Help: "Number of times machine set provisioning has succeded.",
+		}, []string{"name", "namespace"},
+	)
+)
+
+// MachineSetProvisionLabels stores nessesary info about machine set provisioning results
+type MachineSetProvisionLabels struct {
+	Name      string
+	Namespace string
+	Reason    string
+}
+
+func init() {
+	klog.Error("Yay, metrics init!!!!")
+	prometheus.MustRegister(SucceededMachineSetProvisionCount, FailedMachineSetProvisionCount)
```

Comment:
I think controller-runtime normally serves metrics from it's own registry rather than the core prometheus registry doesn't it? 

```
import "sigs.k8s.io/controller-runtime/pkg/metrics"

func init() {
  metrics.Registry.MustRegister(...)
}
```

Reply:
It will eventually go into a single location, and there is no point in importing additional packages from `main.go` to register it to a registry instance from there.

Reply:
Oh, I changed it the way you described without realizing :D